test/server/cards/03_TWI/upgrades/NamelessValor.spec.ts

```diff
@@ -0,0 +1,33 @@
+describe('Nameless Valor', function() {
+    integration(function(contextRef) {
+        it('should only attach to token unit and give it Overwhelm', function() {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    groundArena: [{ card: 'battlefield-marine', damage: 1 }],
+                    hand: ['nameless-valor', 'trade-federation-shuttle'],
+                    leader: { card: 'jyn-erso#resisting-oppression', deployed: true },
+                },
+                player2: {
+                    groundArena: ['coruscant-guard']
+                }
+            });
+            const { context } = contextRef;
+            // create a token for testing
```

Comment:
I just merged the fix that allow token units to be created in `groundArena` so you can do that now if you merge main